### DISCUSSION TITLE
`trio` support

### DISCUSSION THREAD

**Discussion by dhirschfeld-shell** [perspective-dev/perspective#1979] [Ideas]
## Feature Request

My async library of choice is [`trio`](https://github.com/python-trio/trio) rather than `asyncio`. Since `perspective` now supports `starlette` which in turn [supports `trio`](https://github.com/encode/starlette/issues/811) via the [`anyio`](https://github.com/agronholm/anyio) library I'm wondering if it will Just Work, or if there are hard-coded `asyncio` assumptions in `perspective` itself which will prevent that?

If the latter, this issue can be a feature request to replace any hard-coded `asyncio` assumptions with their `anyio` equivalents.

**Comment by dhirschfeld** [comment]
I plan to test this out via trial and error, but it would be good to know if it is expected that `trio` *should* work or is expected to fail.

**Comment by timkpaine** [comment]
this file should be the only one with asyncio expectations https://github.com/finos/perspective/blob/master/python/perspective/perspective/handlers/common.py#L63
https://github.com/finos/perspective/blob/master/python/perspective/perspective/handlers/common.py#L75
https://github.com/finos/perspective/blob/master/python/perspective/perspective/handlers/common.py#L99

**Comment by dhirschfeld** [comment]
Thanks for the info @timkpaine! If it's just that file, it seems feasible to try patching it. If I get something working with `trio` I'll make sure to post back here with my solution.

It may take a little while as it will probably be a case of implementing with `asyncio` first and then porting to ensure I don't break anything...
